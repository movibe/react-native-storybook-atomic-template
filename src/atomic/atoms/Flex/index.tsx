import * as React from 'react'
import { FlexAlignType, ScrollView, StyleProp, TouchableOpacity, View, ViewProps, ViewStyle } from 'react-native'

import { assignTestId } from '../../../utils'

export interface IFlexProps extends ViewProps {
  testID?: string
  reverse?: boolean
  disabled?: boolean
  alignItems?: FlexAlignType
  border?: number
  borderColor?: string
  flex?: number
  onPress?: () => void
  padding?: number
  scroll?: boolean
  content?: boolean
  paddingHorizontal?: number
  paddingVertical?: number
  justifyContent?: 'flex-start' | 'flex-end' | 'center' | 'space-between' | 'space-around' | 'space-evenly'
}

export const Row: React.FC<IFlexProps> = (props) => {
  const {
    reverse,
    alignItems,
    justifyContent,
    style,
    children,
    flex,
    onPress,
    border = 0,
    borderColor,
    padding,
    scroll,
    content,
    disabled,
    testID = 'View',
    ...rest
  } = props
  const direction = reverse ? 'row-reverse' : 'row'
  const _style: StyleProp<ViewStyle> = [
    {
      padding: content ? 10 : padding,
      flexDirection: direction,
      alignItems,
      justifyContent,
      flex,
      borderWidth: border,
      borderColor,
    },
    style,
  ]

  if (onPress) {
    return (
      <TouchableOpacity {...assignTestId('TouchableOpacity', testID)} style={_style} onPress={onPress} disabled={disabled} {...rest}>
        {children}
      </TouchableOpacity>
    )
  }
  if (scroll) {
    return (
      <ScrollView {...assignTestId('ScrollView', testID)} style={_style} {...rest}>
        {children}
      </ScrollView>
    )
  }
  return (
    <View {...assignTestId('View', testID)} style={_style} {...rest}>
      {children}
    </View>
  )
}

export const Column: React.FC<IFlexProps> = (props) => {
  const {
    reverse,
    alignItems,
    justifyContent,
    disabled,
    style,
    children,
    flex,
    onPress,
    content,
    scroll,
    border = 0,
    borderColor,
    padding = 0,
    testID = 'View',
    ...rest
  } = props
  const direction = reverse ? 'column-reverse' : 'column'
  const _style: StyleProp<ViewStyle> = [
    style,
    {
      padding: content ? 10 : padding,
      flexDirection: direction,
      alignItems,
      justifyContent,
      flex,
      borderWidth: border,
      borderColor,
    },
  ]

  if (onPress) {
    return (
      <TouchableOpacity {...assignTestId('TouchableOpacity', testID)} style={_style} onPress={onPress} disabled={disabled} {...rest}>
        {children}
      </TouchableOpacity>
    )
  }
  if (scroll) {
    return (
      <ScrollView {...assignTestId('ScrollView', testID)} style={_style} {...rest}>
        {children}
      </ScrollView>
    )
  }
  return (
    <View {...assignTestId('View', testID)} style={_style} {...rest}>
      {children}
    </View>
  )
}
