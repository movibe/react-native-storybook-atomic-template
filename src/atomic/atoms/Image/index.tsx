import * as React from 'react'
import { Image as Component, ImageProperties } from 'react-native'

import { assignTestId } from '../../../utils'

export interface IImageProps extends ImageProperties {
  testID?: string
}

export const Image: React.FC<IImageProps> = ({ testID = 'image', ...props }): JSX.Element => {
  return <Component {...assignTestId('Image', testID)} {...props} />
}
