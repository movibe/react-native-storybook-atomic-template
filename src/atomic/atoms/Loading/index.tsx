import LottieView from 'lottie-react-native'
import * as React from 'react'
import styled from 'styled-components/native'

import loadingImg from '../../../assets/anim/loader-color.json'
import { assignTestId } from '../../../utils'

interface ContainerProps {
  alignSelf: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'baseline' | 'auto'
}

const Container = styled(LottieView)<ContainerProps>`
  align-self: ${(props) => props.alignSelf};
  flex: 1;
`
const ContainerLoader = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
`

export interface ILoadingProps {
  testID?: string
  autoPlay?: boolean
  loop?: boolean
  speed?: number
  alignSelf?: 'flex-start' | 'flex-end' | 'center' | 'stretch' | 'baseline' | 'auto'
  size?: number | 'small' | 'large'
}

export const Loading: React.FC<ILoadingProps> = ({
  testID = 'Loading',
  alignSelf = 'center',
  autoPlay = true,
  loop = true,
  speed = 1.5,
  size,
  ...props
}): JSX.Element => {
  let style
  switch (size) {
    case 'small':
      style = { width: 25, height: 25 }
      break

    case 'large':
      style = { width: 50, height: 50 }
      break

    default:
      style = {}
  }

  return (
    <ContainerLoader {...assignTestId('View', testID)}>
      <Container alignSelf={alignSelf || 'auto'} source={loadingImg} autoPlay={autoPlay} loop={loop} speed={speed} style={style} {...props} />
    </ContainerLoader>
  )
}
