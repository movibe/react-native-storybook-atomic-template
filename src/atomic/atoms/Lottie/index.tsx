import Animation from 'lottie-react-native'
import * as React from 'react'

import { assignTestId } from '../../../utils'
import { AnimatedLottieViewProps } from './types'

export interface ILottieProps extends AnimatedLottieViewProps {
  id?: string
  testID?: string
}

export const Lottie: React.FC<ILottieProps> = ({ id, testID = 'Lottie', ...props }): JSX.Element => {
  return <Animation {...assignTestId('Lottie', testID)} {...props} />
}
