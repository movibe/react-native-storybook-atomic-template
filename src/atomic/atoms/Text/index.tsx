import * as React from 'react'
import { TextProperties } from 'react-native'
import styled from 'styled-components/native'

import { assignTestId } from '../../../utils'

export interface ITextProps extends TextProperties {
  testID: string
  note?: boolean
  uppercase?: boolean
}

const RNText = styled.Text<ITextProps>`
  color: ${({ theme }) => theme.text.default.color};
  font-size: ${({ theme }) => theme.text.default.size};
  text-transform: ${({ uppercase }) => (uppercase ? 'uppercase' : 'none')};
`

export const Text: React.FC<ITextProps> = ({ testID, children, ...props }): JSX.Element => (
  <RNText {...assignTestId('Text', testID)} {...props}>
    {children}
  </RNText>
)
