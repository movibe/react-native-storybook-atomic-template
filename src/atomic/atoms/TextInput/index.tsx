import * as React from 'react'
import { TextInput as RNTextInput, TextInputProps } from 'react-native'

import { assignTestId } from '../../../utils'

export interface ITextInputProps extends TextInputProps {
  testID?: string
}

export const TextInput: React.FC<ITextInputProps> = ({ testID = 'TextInput', ...props }): JSX.Element => {
  return <RNTextInput {...assignTestId('TextInput', testID)} {...props} />
}
