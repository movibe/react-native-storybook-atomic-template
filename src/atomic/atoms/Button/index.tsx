import * as React from 'react'
import { TextStyle, ViewStyle } from 'react-native'
import styled from 'styled-components/native'

import { assignTestId } from '../../../utils'
import { Text } from '../Text'

const Container = styled.TouchableOpacity.attrs(() => ({
  accessibilityRole: 'link',
}))<IButtonProps>`
  justify-content: center;
  align-items: center;
  border-color: ${({ theme }) => theme.colors.main};
  height: ${({ small }) => (small ? 25 : 45)}px;
  border-radius: ${({ rounded, theme }) => (rounded ? 20 : theme.borderRadius)};
  padding: 0 15px;
  background-color: ${({ dark, color, theme }) => (dark ? theme.colors.dark : color ? theme.colors.main : theme.colors.background)};
  margin-bottom: ${({ marginBottom }) => (marginBottom ? 10 : 0)}px;
  border-width: ${({ border }) => (border ? 1 : 0)};
  opacity: ${({ disabled }) => (disabled ? 0.6 : 1)};
`

interface IText {
  uppercase?: boolean
  dark?: boolean
  textBold?: boolean
  color?: boolean
}

const Content = styled(Text)<IText>`
  color: ${({ color, theme }) => (color ? theme.colors.white : theme.colors.main)};
  font-size: 14px;
  text-align: center;
  text-transform: ${({ uppercase }) => (uppercase ? 'uppercase' : 'none')};
  font-weight: ${({ textBold }) => (textBold ? 800 : 600)};
  font-family: ${({ theme }) => theme.text.font.regular};
`

export interface IButtonProps {
  id?: string
  testID?: string
  text?: string
  textBold?: boolean
  onPress: () => void
  border?: boolean
  borderColor?: boolean
  rounded?: boolean
  marginBottom?: boolean
  color?: boolean
  disabled?: boolean
  full?: boolean
  dark?: boolean
  margin?: boolean
  small?: boolean
  style?: ViewStyle
  customTextStyle?: TextStyle
  customStyle?: ViewStyle
  borderWhite?: boolean
  uppercase?: boolean
  colorBlack?: boolean
  backgroundWhite?: boolean
}

export const Button: React.FC<IButtonProps> = ({
  id,
  testID = 'Button',
  text,
  onPress,
  disabled,
  customStyle,
  customTextStyle,
  uppercase,
  textBold,
  color,
  ...props
}) => {
  return (
    <Container onPress={onPress} style={customStyle} disabled={disabled} color={color} {...props} {...assignTestId('TouchableOpacity', testID)}>
      <Content testID={testID} uppercase={uppercase} textBold={textBold} color={color} style={customTextStyle}>
        {text}
      </Content>
    </Container>
  )
}
