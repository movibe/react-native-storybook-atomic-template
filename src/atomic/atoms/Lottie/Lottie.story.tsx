import * as React from 'react'

import { Lottie } from '../../../atomic/atoms/Lottie'

const size = {
  width: 200,
  height: 200,
}

export default {
  title: 'Atoms/Lottie',
}

export const gift = () => <Lottie source={require('../../../assets/anim/gift.json')} loop autoPlay autoSize style={size} />
export const comment = () => <Lottie source={require('../../../assets/anim/comment.json')} loop autoPlay autoSize style={size} />
export const heart = () => <Lottie source={require('../../../assets/anim/heart.json')} loop autoPlay autoSize style={size} />
export const loaderColor = () => <Lottie source={require('../../../assets/anim/loader-color.json')} loop autoPlay autoSize style={size} />
export const loaderMap = () => <Lottie source={require('../../../assets/anim/loader-map.json')} loop autoPlay autoSize style={size} />
export const loaderMapStar = () => <Lottie source={require('../../../assets/anim/loader-map-star.json')} loop autoPlay autoSize style={size} />
export const loaderMessage = () => <Lottie source={require('../../../assets/anim/loader-message.json')} loop autoPlay autoSize style={size} />
export const loaderPoints = () => <Lottie source={require('../../../assets/anim/loader-points.json')} loop autoPlay autoSize style={size} />
export const loaderPoints2 = () => <Lottie source={require('../../../assets/anim/loader-points2.json')} loop autoPlay autoSize style={size} />
export const loaderPulse = () => <Lottie source={require('../../../assets/anim/loader-pulse.json')} loop autoPlay autoSize style={size} />
export const loaderSuccess = () => <Lottie source={require('../../../assets/anim/loader-success.json')} loop autoPlay autoSize style={size} />
export const loadingSearch = () => <Lottie source={require('../../../assets/anim/loading-search.json')} loop autoPlay autoSize style={size} />
export const stars = () => <Lottie source={require('../../../assets/anim/stars.json')} loop autoPlay autoSize style={size} />
export const upload = () => <Lottie source={require('../../../assets/anim/upload.json')} loop autoPlay autoSize style={size} />
