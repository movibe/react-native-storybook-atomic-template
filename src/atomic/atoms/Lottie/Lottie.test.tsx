import * as React from 'react'
import { render } from 'react-native-testing-library'

import { ILottieProps, Lottie } from '../../../atomic/atoms/Lottie'
import { mockTestID, StorybookHoc } from '../../../utils'

describe('Atom/Lottie', () => {
  let props: ILottieProps = {
    testID: 'animate',
    source: {
      uri: 'https://google.com/animate.json',
    },
  }
  const { toJSON, findByTestId } = render(<Lottie {...props} />, {
    wrapper: StorybookHoc,
  })
  it('should render', async () => {
    const sut = await findByTestId(mockTestID('Lottie', props.testID!).testID)
    expect(sut).toBeDefined()
  })

  it('should render snapshot', () => {
    expect(toJSON()).toMatchSnapshot()
  })
})
