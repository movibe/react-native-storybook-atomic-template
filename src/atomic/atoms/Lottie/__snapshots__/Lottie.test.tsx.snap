// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom/Lottie should render snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100vh",
      },
    ]
  }
>
  <View
    accessible={false}
    style={
      Array [
        undefined,
        Object {
          "alignItems": undefined,
          "borderColor": undefined,
          "borderWidth": 0,
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": undefined,
          "padding": 10,
        },
      ]
    }
    testID="View_View"
  >
    <View
      style={
        Array [
          Object {
            "aspectRatio": NaN,
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          undefined,
        ]
      }
    >
      <View
        accessible={false}
        autoPlay={false}
        enableMergePathsAndroidForKitKatAndAbove={false}
        loop={true}
        onAnimationFinish={[Function]}
        onLayout={[Function]}
        progress={0}
        resizeMode="contain"
        sourceJson="{\\"uri\\":\\"https://google.com/animate.json\\"}"
        speed={1}
        style={
          Object {
            "aspectRatio": NaN,
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="Lottie_animate"
      />
    </View>
  </View>
</View>
`;
